python/dgl/data/wiki_network.py

```diff
@@ -0,0 +1,186 @@
+        super(WikiNetworkDataset, self).__init__(
+            name=name,
+            url=url,
+            raw_dir=raw_dir,
+            hash_key=(url),
+            force_reload=force_reload,
+            verbose=verbose,
+            transform=transform,
+        )
+
+    def process(self):
+        """Load and process the data."""
+        # node features and labels
+        with open(f"{self.raw_path}/out1_node_feature_label.txt", "r") as f:
+            data = f.read().split("\n")[1:-1]
+        features = [
+            [float(v) for v in r.split("\t")[1].split(",")] for r in data
+        ]
+        features = F.tensor(features, dtype=F.data_type_dict["float32"])
+        labels = [int(r.split("\t")[2]) for r in data]
+        self._num_classes = max(labels) + 1
+        labels = F.tensor(labels, dtype=F.data_type_dict["int64"])
+
+        # graph structure
+        with open(f"{self.raw_path}/out1_graph_edges.txt", "r") as f:
+            data = f.read().split("\n")[1:-1]
+            data = [[int(v) for v in r.split("\t")] for r in data]
+        COO = F.swapaxes(F.tensor(data, dtype=F.data_type_dict["int64"]), 0, 1)
+        dst = F.squeeze(F.narrow_row(COO, start=0, stop=1), dim=0)
+        src = F.squeeze(F.narrow_row(COO, start=1, stop=2), dim=0)
+
+        self._g = graph((src, dst), num_nodes=F.shape(features)[0])
+        self._g.ndata["feat"] = features
+        self._g.ndata["label"] = labels
+
+        # split
+        train_masks, val_masks, test_masks = [], [], []
+        for i in range(10):
+            filepath = f"{self.raw_path}/{self.name}_split_0.6_0.2_{i}.npz"
+            f = np.load(filepath)
+            train_masks += [F.zerocopy_from_numpy(f["train_mask"])]
+            val_masks += [F.zerocopy_from_numpy(f["val_mask"])]
+            test_masks += [F.zerocopy_from_numpy(f["test_mask"])]
+        self._g.ndata["train_mask"] = F.astype(
+            F.stack(train_masks, dim=1), F.data_type_dict["bool"]
+        )
+        self._g.ndata["val_mask"] = F.astype(
+            F.stack(val_masks, dim=1), F.data_type_dict["bool"]
+        )
+        self._g.ndata["test_mask"] = F.astype(
+            F.stack(test_masks, dim=1), F.data_type_dict["bool"]
+        )
+
+    def has_cache(self):
+        return os.path.exists(self.raw_path)
+
+    def load(self):
+        self.process()
+
+    def __getitem__(self, idx):
+        assert idx == 0, "This dataset has only one graph."
+        if self._transform is None:
+            return self._g
+        else:
+            return self._transform(self._g)
+
+    def __len__(self):
+        return 1
+
+    @property
+    def num_classes(self):
+        return self._num_classes
+
+
+class ChameleonDataset(WikiNetworkDataset):
+    """Wikipedia page-page network on chameleons from `Multi-scale Attributed
+    Node Embedding <https://arxiv.org/abs/1909.13021>`__ and later processed by
+    `Geom-GCN: Geometric Graph Convolutional Networks
```

Comment:
Why Geom-GCN is related? Is this dataset designated for Geom-GCN?

Reply:
Geom-GCN introduced this variant of the dataset, including turning the task from node regression into node classification, modifying node features, and introducing these dataset splits

Reply:
Maybe reword it. Currently working sounds like this dataset is just used by Geom-GCN.
e.g. later processed by --> introduced by.
```
Wikipedia page-page network on chameleons from `Multi-scale Attributed
    Node Embedding <https://arxiv.org/abs/1909.13021>`__, introduced by
    `Geom-GCN: Geometric Graph Convolutional Networks
```

Reply:
did "later processed by" -> "modified by"